src/taskList.js

```diff
@@ -0,0 +1,55 @@
+import Task from './task.js';
+
+export default class TaskList {
+  constructor(tasks) {
+    this.tasks = tasks.map((el) => new Task(el.description, el.completed, el.index));
+  }
+
+  show() {
+    const toDoList = document.querySelector('.items');
+    toDoList.innerHTML = '';
+    this.tasks.sort((a, b) => a.index - b.index).map((el) => {
+      const task = document.createElement('div');
+      task.classList.add('task');
+      if (el.completed) task.classList.add('completed');
+      task.innerHTML = `<input type="checkbox" ${el.completed ? 'checked' : ''} /> <input type='text' value='${el.description}'/>  <i class="fa fa-ellipsis-v move"></i>`;
+      toDoList.appendChild(task);
+      task.querySelector('input[type="checkbox"]').addEventListener('change', () => {
+        this.toggleTask(el.index);
+      });
+      task.querySelector('input[type="text"]').addEventListener('blur', (e) => {
+        this.tasks[el.index - 1].description = e.target.value;
+        this.save();
+        this.show();
+      });
+      return null;
+    });
+  }
+
+  addTask(task) {
+    task.index = this.tasks.length + 1;
+    this.tasks.push(task);
+    this.show();
+    this.save();
+  }
+
+  toggleTask(i) {
+    this.tasks[i - 1].completed = !this.tasks[i - 1].completed;
+    this.show();
+    this.save();
+  }
+
+  clearCompletedTasks() {
+    this.tasks = this.tasks.filter((el) => el.completed === false);
+    this.tasks.map((el, i) => {
+      el.index = i + 1;
+      return null;
+    });
+    this.show();
+    this.save();
+  }
```

Comment:
- [ ] Good implementation of the `clearCompletedTasks` 👍.  Let's ensure that a function for deleting a task (remove an element from the array) is implemented. Currently, users can't delete a task except they use the `clear all completed` button which is used to delete a completed task. Kindly implement the `delete task` function 💯. Check the requirement again [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#project-requirements)